IPython/core/magics/history.py

```diff
@@ -282,6 +282,7 @@ def recall(self, arg):
                 return
         else:
             self.shell.set_next_input(cmd.rstrip())
+            return
         print("Couldn't evaluate or find in history:", arg)
```

Comment:
I believe that this new return on 285 makes the print on the next line unreachable. With this patch, either you return as part of `except`, or as part of the `else`. So the next question is - under which circumstances _should_ we be printing the "Couldn't evaluate or find n history" line?

Reply:
The `print` is still reachable. Look closely at what L279 does - it skips the L282 `return`, and after exiting the loop you land on L286. I could've improved this whole thing to streamline it and make it readable, but I'm still not sure what is the meaning of the condition on this `if ... : continue`.

Reply:
ah, I see, you're right.